src/server/handlers/github.js

```diff
@@ -321,6 +321,19 @@ const getRequest = (owner, name, token, secret) => {
   };
 };
 
+
+export const hasRepoChanged = async (repositoryUrl, since, token) => {
+  if (since === undefined || !since) {
+    throw new Error('`since` must be an ISO8601 value.');
+  }
+  const { owner, name } = parseGitHubRepoUrl(repositoryUrl);
+  const uri = `/repos/${owner}/${name}/commits?since=${since}`;
+  const response = await requestGitHub.get(uri, { token, json: true });
+  await checkGitHubStatus(response);
+  return response.body.length > 0;
```

Comment:
Yup, seems sufficiently cheap and effective. The individual commit representation is a bit fat, but even if when there are a lot of bulk commits the result is paginated.